sdks/python/apache_beam/runners/dataflow/internal/dependency.py

```diff
@@ -453,36 +457,85 @@ def _build_setup_package(setup_file, temp_dir, build_setup_args=None):
     os.chdir(saved_current_directory)
 
 
-def _stage_beam_sdk_tarball(sdk_remote_location, staged_path, temp_dir):
-  """Stage a Beam SDK tarball with the appropriate version.
+def _desired_sdk_filename_in_staging_location(sdk_location):
+  """Returns the name that SDK file should have file in the staging location.
 
   Args:
-    sdk_remote_location: A GCS path to a SDK tarball or a URL from
-      the file can be downloaded.
-    staged_path: GCS path where the found SDK tarball should be copied.
+    sdk_location: Full path to SDK file.
+  """
+  if sdk_location.endswith('.whl'):
+    if sdk_location.startswith('http'):
+      raise RuntimeError('Staging SDK wheel from an HTTP location is currently'
```

Comment:
good point, this check does not belong here, and actually is useless, since HTTP downloader creates it's own file name after download: https://github.com/apache/beam/blob/007553fc154b058fda5edcebc2ed54100b30f0cc/sdks/python/apache_beam/runners/dataflow/internal/dependency.py#L144

To support download wheels from HTTP we need to save the wheel file under it's original name.  I was not sure if URL download path is used by anyone so didn't change that logic. I can add a TODO there if you think we need to support it.

Reading wheels from GCS is supported with this PR.

Reply:
You can update the JIRA, or open a new one. It is OK to not change the PR now.